src/games/strategy/common/swing/SwingComponents.java

```diff
@@ -91,17 +100,31 @@ public static void promptUser(final String title, final String message, final Ru
 
   public static void promptUser(final String title, final String message, final Runnable confirmedAction,
       final Runnable cancelAction) {
-    SwingUtilities.invokeLater(() -> {
-      int response = JOptionPane.showConfirmDialog(null, message, title,
-          JOptionPane.YES_NO_OPTION, JOptionPane.QUESTION_MESSAGE);
-      boolean result = response == JOptionPane.YES_OPTION;
-
-      if (result) {
-        confirmedAction.run();
-      } else {
-        cancelAction.run();
+
+    boolean showMessage = false;
+    synchronized (visiblePrompts) {
```

Comment:
Swing actions are launched on independent threads, so to prevent the unlikely race condition.